lib/monitor.js

```diff
@@ -214,3 +218,33 @@ internals.forOwn = (obj, func) => {
         func(key, obj[key]);
     }
 };
+
+internals.validateStream = (throwError, lastIndex, index, stream, reporterName, streamName) => {
+
+    if (stream instanceof Stream.Transform) {
+        return internals.assetOrLog(index < lastIndex, `Error in ${reporterName}. ${streamName} is a transform stream and should not be the last stream in the pipeline.`, throwError);
+    }
+
+    if (stream instanceof Stream.Duplex) {
+        return;
+    }
+
+    if (stream instanceof Stream.Writable) {
+        return internals.assetOrLog(index === lastIndex, `Error in ${reporterName}. ${streamName} is a writable stream and should only be the last stream in the pipeline.`, throwError);
+    }
+
+    if (stream instanceof Stream.Readable) {
+        return internals.assetOrLog(false, `Error in ${reporterName}. ${streamName} is a readable stream and should not be present in the pipeline.`, throwError);
+    }
+};
+
+
+internals.assetOrLog = (cond, msg, throwError) => {
```

Comment:
I tried removing the option to throw error.
Unfortunately this makes testing a lot harder since you need to mess with the standard outputs (Otherwise an invalid pipeline with stdin that causes an error will destroy the stream and you wont get any more output out of the test, debugging then becomes a major headache)
I'm sure someone smarter than me can figure a way to get 100% code coverage without messing the outputs but it will take too much time for me to do